### PR TITLE
Fix InitError() and InitWarning() content

### PR DESCRIPTION
**Pull request by hebasto**
Currently, related to wallets and wrong command-line/config option values `InitError()` and `InitWarning()` contents are translated partially: some are translated, others are not.

This PR make translations of all such messages which are meaningful for a user/translator.

Also this PR prevents command-line/config option names from translation in `InitError()` and `InitWarning()` calls (that is not the case in master).

### CODE REVIEW

**Comment by MarcoFalke** [comment]
I'd say some of the errors are exceptions that rarely happen and shouldn't be translated

**Comment by hebasto** [comment]
@MarcoFalke added rationale, removed [WIP]

**Comment by laanwj** [comment]
> I'd say some of the errors are exceptions that rarely happen and shouldn't be translated

I agree. There's also the problem, especially with extremely rare error messages (such as the secp256k1 failure) that translating them makes them impossible to google. This doesn't really weigh up to the advantage of translating the message if it's very technical in the first place.

So I'm not convinced this is a good thing.

**Comment by hebasto** [comment]
@laanwj 

> There's also the problem, especially with extremely rare error messages (such as the secp256k1 failure) that translating them makes them impossible to google.

Agree. Should 128148775ee465047c2a9c0e2a92277981b1a4b7 commit be removed completely?

**Comment by laanwj** [comment]
I'd say yes.
(coincidentally, too late for this for 0.18, translation string freeze was February 1)

**Comment by hebasto** [comment]
@laanwj 
> (coincidentally, too late for this for 0.18, translation string freeze was February 1)

I know it from the last IRC meeting. 0.18 is not the target for this PR :)

**Comment by hebasto** [comment]
Ref: [IRC discussion](http://www.erisian.com.au/bitcoin-core-dev/log-2019-02-02.html#l-265)

**Comment by hebasto** [comment]
> I'd say some of the errors are exceptions that rarely happen and shouldn't be translated

Removed translations of error messages inside `InitSanityCheck()`.
PR description updated.

**Comment by hebasto** [comment]
[Translation Strings Policy](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_strings_policy.md#gui-strings):
> Anything that appears to the user in the GUI is to be translated. This includes labels, menu items, button texts, tooltips and window titles. This includes messages passed to the GUI through the UI interface through **`InitMessage`**, `ThreadSafeMessageBox` or `ShowProgress`.

**Comment by hebasto** [comment]
Ref: [comment](https://github.com/bitcoin/bitcoin/pull/6870#discussion_r42965071) by @laanwj:

> I think errors relating directly to user options should be kept translated.

**Comment by hebasto** [comment]
Rebased.

**Comment by hebasto** [comment]
Rebased.

File: src/init.cpp [hunk 1]
```diff
@@ -1384,7 +1383,7 @@ bool AppInitMain(InitInterfaces& interfaces)
         for (const std::string& snet : gArgs.GetArgs("-onlynet")) {
             enum Network net = ParseNetwork(snet);
             if (net == NET_UNROUTABLE)
-                return InitError(strprintf(_("Unknown network specified in -onlynet: '%s'"), snet));
+                return InitError(strprintf(_("Unknown network specified in %s: '%s'"), "-onlynet", snet));
```
Comment (PastaPastaPasta) [inline src/init.cpp]: bodies of an if statement should be on the same line or in brackets, imo this could be fixed in this PR

File: src/init.cpp [hunk 2]
```diff
@@ -1429,11 +1428,11 @@ bool AppInitMain(InitInterfaces& interfaces)
         } else {
             CService onionProxy;
             if (!Lookup(onionArg.c_str(), onionProxy, 9050, fNameLookup)) {
-                return InitError(strprintf(_("Invalid -onion address or hostname: '%s'"), onionArg));
+                return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-onion", onionArg));
             }
             proxyType addrOnion = proxyType(onionProxy, proxyRandomize);
             if (!addrOnion.IsValid())
-                return InitError(strprintf(_("Invalid -onion address or hostname: '%s'"), onionArg));
+                return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-onion", onionArg));
```
Comment (PastaPastaPasta) [inline src/init.cpp]: same as above, same line or brackets

File: src/init.cpp [hunk 3]
```diff
@@ -1829,7 +1828,7 @@ bool AppInitMain(InitInterfaces& interfaces)
         CSubNet subnet;
         LookupSubNet(net.c_str(), subnet);
         if (!subnet.IsValid())
-            return InitError(strprintf(_("Invalid netmask specified in -whitelist: '%s'"), net));
+            return InitError(strprintf(_("Invalid netmask specified in %s: '%s'"), "-whitelist", net));
```
Comment (PastaPastaPasta) [inline src/init.cpp]: same as above, same line or brackets

File: src/wallet/init.cpp [hunk 4]
```diff
@@ -115,14 +115,14 @@ bool WalletInit::ParameterInteraction() const
...
     if (is_multiwallet) {
         if (gArgs.GetBoolArg("-upgradewallet", false)) {
-            return InitError(strprintf("%s is only allowed with a single wallet file", "-upgradewallet"));
+            return InitError(strprintf(_("%s is only allowed with a single wallet file"), "-upgradewallet"));
         }
     }
 
     if (gArgs.GetBoolArg("-sysperms", false))
-        return InitError("-sysperms is not allowed in combination with enabled wallet functionality");
+        return InitError(strprintf(_("%s is not allowed in combination with enabled wallet functionality"), "-sysperms"));
```
Comment (PastaPastaPasta) [inline src/wallet/init.cpp]: same line or brackets

File: src/wallet/init.cpp [hunk 5]
```diff
@@ -115,14 +115,14 @@ bool WalletInit::ParameterInteraction() const
...
     if (gArgs.GetArg("-prune", 0) && gArgs.GetBoolArg("-rescan", false))
-        return InitError(_("Rescans are not possible in pruned mode. You will need to use -reindex which will download the whole blockchain again."));
+        return InitError(strprintf(_("Rescans are not possible in pruned mode. You will need to use %s which will download the whole blockchain again."), "-reindex"));
```
Comment (PastaPastaPasta) [inline src/wallet/init.cpp]: same line or brackets

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/init.cpp [hunk 6]
```diff
@@ -917,7 +917,7 @@ bool AppInitBasicSetup()
 #endif
 
     if (!SetupNetworking())
-        return InitError("Initializing networking failed");
+        return InitError(_("Initializing networking failed"));
```
Comment (MarcoFalke) [inline src/init.cpp]: Tend to NACK. This will put the translated message into the debug log

Comment (MarcoFalke) [reply src/init.cpp]: I guess the translation should be registered here only and then later translated in InitError with `_()`

**Comment by hebasto** [comment]
@MarcoFalke
> Tend to NACK. This will put the translated message into the debug log

> I guess the translation should be registered here only and then later translated in InitError with `_()`

Agree. There was a dedicated PR #15340.

So, closing this PR now.

**Pull request closed without merging**